### ISSUE TITLE
How to design and build for low bandwidth

### ISSUE THREAD

**Issue by kristinashu** [mozilla/coss#95]
I will make design notes on what we need to show vs what is bonus content (e.g. images).

Can we do a query to see what bandwidth a user has and then give them the appropriate version of our site?

cc @Pomax @alanmoo

**Comment by alanmoo** [comment]
Nope, bandwidth queries are notoriously not-a-thing. There are many ways to prepare for low-bandwidth user experiences, but directly checking for it is not one of them.

**Comment by Pomax** [comment]
progressive enhancements are pretty much the only "known" solution at the moment

**Comment by a-kilroy** [comment]
This basically means we can determine the order in which things load so that critical content is prioritized. Did I express this well?

**Comment by kristinashu** [comment]
Sounds good to me.

**Comment by alanmoo** [comment]
Along those lines, yes. Things like offering multiple image sources (low res, high res, etc), reducing HTTP requests, and only loading content that's most likely to be seen (don't load all the images at once, etc.)

**Comment by m-branson** [comment]
Re: content loading - [lazy loading](https://en.wikipedia.org/wiki/Lazy_loading)?

And does mezzanine have something like [image transforms](https://craftcms.com/docs/image-transforms) available? Including a focal point on images when they're uploaded will help the transforms be most affective as well (see #90).

**Comment by alanmoo** [comment]
Yup, lazy loading, but used judiciously. I'm not sure about image transforms, I've used a service to handle that before (I think it was [Cloudinary](http://cloudinary.com/)). Depending on what we're trying to do, that might be worth separately investigating.
